src/runtime_src/core/common/sensor.cpp

```diff
@@ -16,8 +13,248 @@
+  sensor_array.push_back(std::make_pair("", 
+    populate_sensor<xrt_core::query::noop, xrt_core::query::v12_in_aux1_i_milliamps>(device, "v12_in_aux1_i", "V12 in Aux1 i")));
+  sensor_array.push_back(std::make_pair("", 
+    populate_sensor<xrt_core::query::vcc_aux_millivolts, xrt_core::query::noop>(device, "vcc_aux", "Vcc Auxillary")));
+  sensor_array.push_back(std::make_pair("", 
+    populate_sensor<xrt_core::query::vcc_aux_pmc_millivolts, xrt_core::query::noop>(device, "vcc_aux_pmc", "Vcc Auxillary Pmc")));
+  sensor_array.push_back(std::make_pair("", 
+    populate_sensor<xrt_core::query::vcc_ram_millivolts, xrt_core::query::noop>(device, "vcc_ram", "Vcc Ram")));
+  sensor_array.push_back(std::make_pair("", 
+    populate_sensor<xrt_core::query::v0v9_int_vcc_vcu_millivolts, xrt_core::query::noop>(device, "0v9_vccint_vcu", "0.9 Volts Vcc Vcu")));
+  root.add_child("power_rails", sensor_array);
+  return root;
+}
+
+
+boost::property_tree::ptree
+read_thermals(const xrt_core::device * device)
+{
+  boost::property_tree::ptree root;
+  boost::property_tree::ptree thermal_array;
+  //--- pcb ----------
+  thermal_array.push_back(std::make_pair("", populate_temp<xrt_core::query::temp_card_top_front>(device, "pcb_top_front", "PCB Top Front")));
+  thermal_array.push_back(std::make_pair("", populate_temp<xrt_core::query::temp_card_top_rear>(device, "pcb_top_rear", "PCB Top Rear")));
+  thermal_array.push_back(std::make_pair("", populate_temp<xrt_core::query::temp_card_bottom_front>(device, "pcb_bottom_front", "PCB Bottom Front")));
+
+  //--- cage ----------
+  thermal_array.push_back(std::make_pair("", populate_temp<xrt_core::query::cage_temp_0>(device, "cage_temp_0", "Cage0")));
+  thermal_array.push_back(std::make_pair("", populate_temp<xrt_core::query::cage_temp_1>(device, "cage_temp_1", "Cage1")));
+  thermal_array.push_back(std::make_pair("", populate_temp<xrt_core::query::cage_temp_2>(device, "cage_temp_2", "Cage2")));
+  thermal_array.push_back(std::make_pair("", populate_temp<xrt_core::query::cage_temp_3>(device, "cage_temp_3", "Cage3")));
+
+  // --- fpga, vccint, hbm -------------
+  thermal_array.push_back(std::make_pair("", populate_temp<xrt_core::query::temp_fpga>(device, "fpga0", "FPGA")));
+  thermal_array.push_back(std::make_pair("", populate_temp<xrt_core::query::int_vcc_temp>(device, "int_vcc", "Int Vcc")));
+  thermal_array.push_back(std::make_pair("", populate_temp<xrt_core::query::hbm_temp>(device, "fpga_hbm", "FPGA HBM")));
+
+  root.add_child("thermals", thermal_array);
+  return root;
+}
+
+boost::property_tree::ptree
+read_power_consumption(const xrt_core::device * device)
+{
+  boost::property_tree::ptree root;
+  boost::property_tree::ptree power_array;
+  std::string power_watts;
+  try {
+    auto power_level = xrt_core::device_query<xrt_core::query::max_power_level>(device);
+    power_watts = lvl_to_power_watts(power_level);
+  }
+  catch (...) {
+    power_watts = "N/A";
```

Comment:
Hmm...this is scary for it could hide a true issue.  

Is there some other way to determine if this value is not valid.